siliconcompiler/schema/schema_obj.py

```diff
@@ -235,16 +229,52 @@ def clear(self, *keypath):
         if not Schema._is_leaf(cfg):
             raise ValueError(f'Invalid keypath {keypath}: clear() must be called on a complete keypath')
 
+        err = Schema._validate_step_index(cfg['pernode'], 'value', step, index)
+        if err:
+            raise ValueError(f'Invalid args to clear() of keypath {keypath}: {err}')
+
         if cfg['lock']:
             # TODO: log here
             return False
 
-        cfg['value'] = cfg['defvalue']
-        cfg['nodevalue'] = {}
-        cfg['set'] = False
+        if step is None and index is None:
+            cfg['nodevalue'] = {}
+            cfg['value'] = cfg['defvalue']
+            cfg['set'] = False
+        else:
+            if index is None:
+                index = 'default'
+            try:
+                del cfg['nodevalue'][step][index]
+            except KeyError:
+                # If this key doesn't exist, silently continue - it was never set
+                pass
 
         return True
 
+    def getvals(self, *keypath):
+        """
+        Returns all values (global and pernode) associated with a particular parameter.
+
+        Returns a list of tuples of the form (value, step, index).
+        """
+        cfg = self._search(*keypath)
+
+        if not Schema._is_leaf(cfg):
+            raise ValueError(f'Invalid keypath {keypath}: getvals() must be called on a complete keypath')
+
+        vals = []
+        if cfg['pernode'] != 'required':
+            vals.append((self.get(*keypath, step=None, index=None), None, None))
```

Comment:
Could it make sense to order the output as `(step, index, value)` instead of `(value, step, index)`?

On the one hand, we are putting `step=`/`index=` at the end of function signatures since they're keyword arguments.

But in the schema, we usually do `chip.set([...], step, index, value)`.

Reply:
That's a nice suggestion, I see both sides of this and don't have a particularly strong preference. @gadfort any thoughts?

I am making `_getvals()` private for the time being, so I do feel less urgent about getting this right now.